mlrun/projects/pipelines.py

```diff
@@ -65,6 +65,15 @@ def get_workflow_engine(engine_kind, local=False):
     )
 
 
+def generate_text_for_workflow(run_id, errors_counter, state):
```

Comment:
Can be a staticmethod of `_PipelineRunner`
```suggestion
def _generate_workflow_finished_message(run_id, errors_counter, state):
```